internal/tunnel/tunnel.go

```diff
@@ -62,8 +62,13 @@ type Tunnel interface {
 	// SOCKS5ProxyURL returns the SOCSK5 proxy URL.
 	SOCKS5ProxyURL() *url.URL
 
+	// LogFilePath returns the location of the logfile. If
+	// you read the log file before calling stop, you should
+	// be prepared for it to contain incomplete lines.
+	LogFilePath() (string, bool)
+
```

Comment:
Tor was already saving a logfile. So, what we need here is to expose it. All tunnels but tor will not have a log file for now. But we'll probably also add more detailed logs for psiphon in the future.